errors/mappers/validationerrors/interceptor.go

```diff
@@ -0,0 +1,105 @@
+package validationerrors
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"google.golang.org/grpc"
+)
+
+type validator interface {
+	Validate() error
+}
+
+// UnaryServerInterceptor returns a new unary server interceptor that validates incoming messages.
+//
+// Invalid messages will be rejected with `InvalidArgument` before reaching any userspace handlers.
+func UnaryServerInterceptor() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		if v, ok := req.(validator); ok {
+			if err := v.Validate(); err != nil {
+				// Fill ValidationError and throw it forward
+				return nil, GetValidationError(err)
+			}
+		}
+		return handler(ctx, req)
+	}
+}
+
+// GetValidationError function returns a validation error from an error.
+func GetValidationError(err error) error {
+	// Check if the error is of type of validation
+	typeOfError := reflect.TypeOf(err)
+	valueOfError := reflect.ValueOf(err)
+
+	// Check if the error is a struct and a type of validation error that contains a cause.
+	if valueOfError.Kind() == reflect.Struct && strings.Contains(typeOfError.String(), "ValidationError") && !valueOfError.FieldByName("Cause").IsNil() {
+		cause := valueOfError.FieldByName("Cause")
+		typeOfCause := reflect.TypeOf(cause.Interface())
+
+		valueOfCause := reflect.ValueOf(cause.Interface())
+		if strings.Contains(typeOfCause.String(), "ValidationError") && valueOfCause.FieldByName("Field").String() != "" && valueOfCause.FieldByName("Reason").String() != "" && !valueOfCause.FieldByName("Cause").IsNil() {
```

Comment:
Might be nice to break this up into a few lines that are shorter because it extends pretty far off the page. I think [the 80 character limit](https://softwareengineering.stackexchange.com/questions/312889/how-to-follow-the-80-character-limit-best-practice-while-writing-source-code) is a good rule of thumb, but it's not something to obsess over.